sitemaps-index.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Class WP_Sitemaps_Index.
+ * Builds the sitemap index page that lists the links to all of the sitemaps.
+ *
+ * @todo This will probably be split out so that rewrites are in a class, building the xml output is a class,
+ * rendering sitemaps content is a class etc.
+ */
+class WP_Sitemaps_Index {
+
+	/**
+	 * Content of the sitemap to output.
+	 *
+	 * @var string
+	 */
+	protected $sitemap_content = '';
+
+	/**
+	 * Class constructor.
+	 */
+	public function __construct() {
+		add_action( 'init', array( $this, 'url_rewrites' ), 99 );
+		add_filter( 'redirect_canonical', array( $this, 'redirect_canonical' ) );
+		add_action( 'template_include', array( $this, 'output_sitemap' ) );
+	}
+
+	/**
+	 * Sets up rewrite rule for sitemap_index.
+	 * @todo Additional rewrites will probably need adding to this.
+	 */
+	public function url_rewrites() {
+		global $wp;
+		$wp->add_query_var( 'sitemap' );
```

Comment:
Instead of using the global in this way, the following code can be used that has more safety checks around it.
`add_rewrite_tag('%sitemap%','sitemap');`